grc/gui/Executor.py

```diff
@@ -83,8 +84,14 @@ def run(self):
         r = "\n"
         while r:
             GLib.idle_add(Messages.send_verbose_exec, r)
-            r = self.process.stdout.read(1024).decode('utf-8','ignore')
-        self.process.poll()
+            r = self.process.stdout.read(1)
+
+        # Properly close pipe before thread is terminated
+        self.process.stdout.close()
+        while self.process.poll() is None:
```

Comment:
This is a definitive improvement, ideally it would have a timeout value.

Reply:
I mostly agree, but we are sending a kill signal not a term signal, so if we were to exit after a time-out the process would still be running.  I would be more inclined to add the logic to send terminate() and then after a timeout fall-back to a kill.   Python 3 wait take a timeout option, so we could use that instead of poll, but we would have to implement a not great workaround for Python 2.